jerry-core/vm/vm.c

```diff
@@ -382,6 +382,22 @@ vm_get_implicit_this_value (ecma_value_t *this_value_p) /**< [in,out] this value
   return false;
 } /* vm_get_implicit_this_value */
 
+
+/**
+ * Special bytecode sequence for error handling while the vm_loop
+ * is preserved for an execute operation
+ */
+static const uint8_t vm_error_byte_code_p[] =
+{
+  CBC_EXT_OPCODE, CBC_EXT_ERROR
+};
+
+/**
+ * Location of the currently executed byte code if an
+ * error occours while the vm_loop is preserved
+ */
+static uint8_t *vm_error_byte_code_location_p = NULL;
```

Comment:
Why do we exactly need this? Could you show me an example?

Reply:
[This commit ](https://github.com/rerobika/jerryscript/compare/75b778c905e5...c8bdc4da53d9)is the same as the PR's content but without the usage of this static variable.

It fails on the `debugger/do_exception.js`. See the details [here](https://travis-ci.org/rerobika/jerryscript/jobs/453856622#L564).

Reply:
This is good news! So we only need this variable when the debugger is enabled to determine the precise location of the exception. This could be moved inside the debugger part of the JCONTEXT.

Reply:
One more thing: when the debugger stops because of an exception, the byte code pointer should point the offending byte code. If the debugger sees the byte code after the call, it might be part of a different JavaScript expression.